isis/src/base/apps/framestitch/framestitch.cpp

```diff
@@ -189,34 +201,81 @@ namespace Isis {
+    // output cube if requested.
     if (ui.GetBoolean("FLIP")) {
+      // Need a temporary buffer to help with the swapping of top and
+      // bottom portions of outCube.
       Brick topBuff(outCube->sampleCount(), frameHeight, outCube->bandCount(), outCube->pixelType());
-      Brick bottomBuff(outCube->sampleCount(), frameHeight, outCube->bandCount(), outCube->pixelType());
-      Brick tempBuff(outCube->sampleCount(), frameHeight, outCube->bandCount(), outCube->pixelType());
+      Brick botBuff(outCube->sampleCount(), frameHeight, outCube->bandCount(), outCube->pixelType());
+      Brick tmpBuff(outCube->sampleCount(), frameHeight, outCube->bandCount(), outCube->pixelType());
 
       for (int frame = 0; frame < numFrames / 2; frame++) {
         topBuff.SetBasePosition(1,frame * frameHeight + 1,1);
         outCube->read(topBuff);
-        bottomBuff.SetBasePosition(1,outCube->lineCount() - (frame + 1) * frameHeight + 1,1);
-        outCube->read(bottomBuff);
+        botBuff.SetBasePosition(1,outCube->lineCount() - (frame + 1) * frameHeight + 1,1);
+        outCube->read(botBuff);
 
-        tempBuff.Copy(topBuff);
-        topBuff.Copy(bottomBuff);
+        tmpBuff.Copy(topBuff);
+        topBuff.Copy(botBuff);
         outCube->write(topBuff);
-        bottomBuff.Copy(tempBuff);
-        outCube->write(bottomBuff);
+        botBuff.Copy(tmpBuff);
+        outCube->write(botBuff);
       }
 
       if (!outInst.hasKeyword("DataFlipped")) {
         outInst.addKeyword(PvlKeyword("DataFlipped"));
       }
       outInst["DataFlipped"].setValue(toString(!inputFlipped));
     }
+  
+    // Wipe numLinesOverlap/2 lines at the top and same amount at the bottom
+    // of each framelet to reduce their overlap.
 
-    process.EndProcess();
+    Brick srcBuff(outCube->sampleCount(), 1, outCube->bandCount(), outCube->pixelType());
+    Brick dstBuff(outCube->sampleCount(), 1, outCube->bandCount(), outCube->pixelType());
+    
+    for (int line = 0; line < outCube->lineCount(); line++) {
+      int frame = line / frameHeight; // block index
+      int lineInFrame = line - frameHeight * frame; // line index in the current block
+
+      // Skip these lines, which will result in them being wiped
+      if (lineInFrame < numLinesOverlap / 2 || lineInFrame >= frameHeight - numLinesOverlap / 2)
+        continue;
+
+      // Same book-keeping is needed to compute where the line will go
+      int src = line;
+      int dst = lineInFrame - numLinesOverlap / 2 + (frameHeight - numLinesOverlap) * frame;
 
+      srcBuff.SetBasePosition(1, src + 1, 1);
+      dstBuff.SetBasePosition(1, dst + 1, 1);
+
+      outCube->read(srcBuff);  // read into srcBuf
+      dstBuff.Copy(srcBuff);   // copy from srcBuff to dstBuff
+      outCube->write(dstBuff); // write into outCube
+    }
+    
+    // Wipe the last lines of the image after they are transferred
+    // TODO: Figure out how to resize the cube.
```

Comment:
The resize needs to be done before any data is copied to it. Ideally this is done when we call process.SetOutputCube

Reply:
When it comes to tests, will it be enough to just tweak the tests you added when this tool was added? I did not work with ISIS tests before. I assume there's no need for me to add new files, but likely just some more lines of code where appropriate in the "Fixtures" code. Is that correct?

If the cub file is resized, the number of lines in it won't correspond to what the CSM model thinks the number of lines is. Is that a problem? Likely not, but may be confusing. In the CSM model there is logic for having the reduced number of lines, which would agree with the number or actual lines in the cub file. 

> The number of lines trimmed out should be added to the image label next to the information about flipping. This way we can pull it into the CSM model later.

OK. This will need some logic changes in the CSM model as well then. I will get to this tomorrow.

Reply:
The existing tests will need updating because of the default 2 line trim breaking them. You'll also need to add a test that sets the trimming and then checks that it's doing the correct thing.

I don't think the fixtures will need to be updated as they handle input data. These tests are where the updates will need to go https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/tests/FunctionalTestsFrameStitch.cpp

If you have any questions or anything let me know. I can get on a call and go over some stuff or we can just go back and forth here.

Reply:
I fixed the tests, by setting to 0 the number of lines to trim by default, as for a general application that should be the case. For WAC no single number works all the time, it has to be 2 or 4 depending on the cub.

I added a test where the number of lines to trim is 2.

The cube is resized on writing, so no empty lines are written.

The number of lines that were trimmed are written to the output cube. Next step will be making changes to ALE and do CSM to query this. 

I tested this very carefully with two cub files and the result is as before, except the intended changes as above.

Reply:
Here's a problem though. This tool will write the number of images lines to be trimmed in the stitched cube. But later, when ALE reads in this cube, it won't have kernels. So ALE needs to read a cube before being stitched, but that won't have the number of image lines to trimp. 

Can framestitch write the kernels so it is an honest cube?

Reply:
I strip the kernels off because the stitched image is no longer valid for the ISIS camera model. I could try something else to prevent it from creating a camera model.